flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/UnionITCase.scala

```diff
@@ -69,6 +69,36 @@ class UnionITCase(
   }
 
   @Test
+  def testUnionAllDifferentFieldNames(): Unit = {
```

Comment:
We are trying to reduce the number of ITCases in of the Table API and only add new ones if necessary to keep the build time low.

I would change the existing tests to use different field names instead of adding new tests here. In addition the tests should try to select a field to check that the field names of the first input are forwarded.